### PR TITLE
feat(webdriver): implement page.setCacheEnabled

### PR DESCRIPTION
**Pull request by OrKoN**
This PR implements page.setCacheEnabled for WebDriver BiDi.

### CODE REVIEW

File: packages/puppeteer-core/src/bidi/core/BrowsingContext.ts [hunk 1]
```diff
@@ -417,6 +417,18 @@ export class BrowsingContext extends EventEmitter<{
     });
   }
 
+  @throwIfDisposed<BrowsingContext>(context => {
+    // SAFETY: Disposal implies this exists.
+    return context.#reason!;
+  })
+  async setCacheBypass(enabled: boolean): Promise<void> {
+    // @ts-expect-error not in BiDi types yet.
+    await this.#session.send('browsingContext.setCacheBypass', {
```
Comment (juliandescottes) [inline packages/puppeteer-core/src/bidi/core/BrowsingContext.ts]: Our command is in the network module
```suggestion
    await this.#session.send('network.setCacheBypass', {
```

**Review by juliandescottes — COMMENTED** [review]
Quick flyby

File: packages/puppeteer-core/src/bidi/Page.ts [hunk 2]
```diff
@@ -537,6 +537,10 @@ export class BidiPage extends Page {
   }
 
   override async setCacheEnabled(enabled?: boolean): Promise<void> {
+    if (!this.#browserContext.browser().cdpSupported) {
+      await this.#frame.browsingContext.setCacheBypass(enabled ?? false);
```
Comment (juliandescottes) [inline packages/puppeteer-core/src/bidi/Page.ts]:
```suggestion
      await this.#frame.browsingContext.setCacheBypass(!enabled ?? true);
```

Comment (OrKoN) [reply packages/puppeteer-core/src/bidi/Page.ts]: did you mean to change to `enabled ?? true`?

Comment (OrKoN) [reply packages/puppeteer-core/src/bidi/Page.ts]: ah sorry, you are totally write, not sure what I was reading :D

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**